include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,508 @@
+    //VkBool32           shaderStorageImageArrayNonUniformIndexing;
+    //VkBool32           shaderInputAttachmentArrayNonUniformIndexing;
+    //VkBool32           shaderUniformTexelBufferArrayNonUniformIndexing;
+    //VkBool32           shaderStorageTexelBufferArrayNonUniformIndexing;
+    //VkBool32           descriptorBindingUniformBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingSampledImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageImageUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUniformTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingStorageTexelBufferUpdateAfterBind;
+    //VkBool32           descriptorBindingUpdateUnusedWhilePending;
+    //VkBool32           descriptorBindingPartiallyBound;
+    //VkBool32           descriptorBindingVariableDescriptorCount;
+    //VkBool32           runtimeDescriptorArray;
+    //VkBool32           samplerFilterMinmax;
+    //VkBool32           scalarBlockLayout;
+    //VkBool32           imagelessFramebuffer;
+    //VkBool32           uniformBufferStandardLayout;
+    //VkBool32           shaderSubgroupExtendedTypes;
+    //VkBool32           separateDepthStencilLayouts;
+    //VkBool32           hostQueryReset;
+    //VkBool32           timelineSemaphore;
+    bool bufferDeviceAddress = false;
+    //VkBool32           bufferDeviceAddressCaptureReplay;
+    //VkBool32           bufferDeviceAddressMultiDevice;
+    //VkBool32           vulkanMemoryModel;
+    //VkBool32           vulkanMemoryModelDeviceScope;
+    //VkBool32           vulkanMemoryModelAvailabilityVisibilityChains;
+    //VkBool32           shaderOutputViewportIndex;
+    //VkBool32           shaderOutputLayer;
+    //VkBool32           subgroupBroadcastDynamicId;
+
+    /* Vulkan 1.3 Core */
+    //VkBool32           robustImageAccess;
+    //VkBool32           inlineUniformBlock;
+    //VkBool32           descriptorBindingInlineUniformBlockUpdateAfterBind;
+    //VkBool32           pipelineCreationCacheControl;
+    //VkBool32           privateData;
+    //VkBool32           shaderDemoteToHelperInvocation;
+    //VkBool32           shaderTerminateInvocation;
+    //VkBool32           subgroupSizeControl;
+    //VkBool32           computeFullSubgroups;
+    //VkBool32           synchronization2;
+    //VkBool32           textureCompressionASTC_HDR;
+    //VkBool32           shaderZeroInitializeWorkgroupMemory;
+    //VkBool32           dynamicRendering;
+    //VkBool32           shaderIntegerDotProduct;
+    //VkBool32           maintenance4; -> Doesn't make sense to expose, too vulkan specific
+
+    /* CoherentMemoryFeaturesAMD *//* VK_AMD_device_coherent_memory */
+    //VkBool32           deviceCoherentMemory;
+
+    /* VK_AMD_shader_early_and_late_fragment_tests *//* couldn't find the struct/extension in latest vulkan headers */
```

Comment:
`VkBool32 shaderEarlyAndLateFragmentTests;`

Reply:
no idea how to use it in GLSL, don't expose yet

Reply:
its a shader-only extension so you can just enable it and report as a limit (since it needs an explicitl opt-in in the shader coe anyway)